### PR TITLE
core: add buckets property to regionInfo struct

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?
This is the first step for the final storage of buckets in memery. 
Some api and storage depences on it.
Issue Number: Relate #4668

### What is changed and how it works?

```commit-message
```

### Check List

Tests

Code changes

Side effects

Related changes

### Release note

```release-note
 None.
```

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -214,6 +215,14 @@ func (r *RegionInfo) Clone(opts ...RegionCreateOption) *RegionInfo {
 	return region
 }
 
+// GetBuckets returns the buckets of the region.
+func (r *RegionInfo) GetBuckets() *metapb.Buckets {
+	if r == nil {
```
Comment (rleungx) [inline server/core/region.go:220]: When will it happen?

Comment (bufferflies) [reply server/core/region.go:220]: this is the first step,  sync and some api will use it, I will split them into different pr.

Comment (rleungx) [reply server/core/region.go:220]: I mean `r == nil`

**Pull request closed without merging**